lib/os/reboot.c

```diff
@@ -4,10 +4,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <reboot.h>
 #include <kernel.h>
 #include <sys/printk.h>
+#include <toolchain.h>
 
-extern void sys_arch_reboot(int type);
+extern FUNC_NORETURN void sys_arch_reboot(int type);
```

Comment:
The `FUNC_NORETURN` qualifier here is a no-op in practice because this is not globally defined and does not propagate to the files where the function implementations are declared.

Also, `sys_arch_reboot` may not necessarily be a "no return" function -- most `sys_arch_reboot` implementations currently set some "reset" register value or call a HAL function to cause reset. If we add `FUNC_NORETURN` for it, we will need to add an infinite loop at the end of all of these functions to prevent compiler warning.

Leaving `sys_arch_reboot` as is can be justified for the above reason and also that `sys_reboot` catches any failure to reset (for whatever reason) and prints out an error message.